### PR TITLE
Switch to using GitHub Actions for testing the site

### PR DESCRIPTION
**Pull request by willfurnass**
- [x] Remove Travis config
- [x] Remove mention of Travis from the README
- [x] Add GHA workflow that builds the site just for PRs
- [x] Add *commented-out* GHA workflow tasks that perform internal link checking.  Can uncomment once confident that won't block all future PRs.  Previous attempt to (re-)-enable internal ink checking: #240

### CODE REVIEW

**Comment by willfurnass** [comment]
Might need to merge this PR and generate a trivial PR to test this... :(

**Comment by davidwyld** [comment]
> Might need to merge this PR and generate a trivial PR to test this... :(

Or we could just add:
```
push:
    branches: gha-ci-tests
```
and check it runs the actions correctly on your repo?

**Comment by willfurnass** [comment]
Good thinking!  GHA status from this will be reported in my fork: https://github.com/willfurnass/RSE-Sheffield.github.io/actions/runs/295019372

**Comment by willfurnass** [comment]
@davidwilby The triggered GHA job passed so I think we can merge this if you're happy.

However, I think this workflow (and the previous Travis workflow) are of limited value until we get automated internal link checking working (Issue #239)

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: CI build 
+on:
+  # push:
+  #   branches:
+  #     - master
+  pull_request:
+    branches:
+      - master
+  push:
+    branches: gha-ci-tests
```
Comment (davidwyld) [inline .github/workflows/ci-build.yaml]: Looks good, just need to remove this line before merging.

**Review by davidwyld — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yaml (hunk 1 above)
Comment (willfurnass) [reply .github/workflows/ci-build.yaml]: Done

**Review by davidwyld — APPROVED** [review]

**Comment by davidwyld** [comment]
Ha, I can't merge it because the travis checks haven't (and won't) run...

image

@willfurnass  Can you merge with your admin privileges?

**Comment by willfurnass** [comment]
FYI I've removed the Travis webhook from this repo and removed Travis integration from the entire Org.

**Comment by willfurnass** [comment]
Force-merged as admin.

**Pull request merged**